### ISSUE TITLE
Last manually-closed remote workspace restores last local workspace on startup

### ISSUE THREAD

**Issue by noinfo** [zed-industries/zed#7759]
Summary (quoting @notpeter): If you close the workspace (cmd+w) before quitting (cmd+q) when you come back, Zed reopens that workspace instead of giving you a blank window. This behavior differs what other editors do and thus what some users will expect.

### Describe the bug / provide steps to reproduce it

I have downloaded zed and opened a folder as a project. When I close the project I habe zed in the state that I want it in on startup. However whenever I start up zed it opens the closed project again. 

I can not close the project and have it stay closed. It always reopens on its own.

**Comment by williamviktorsson** [comment]
Are you looking for something different than setting ` "restore_on_startup": "none"` in your settings.json?

**Comment by noinfo** [comment]
Yes, I think I do. When I actively remove any open projects from zed before closing, I expect it to open "empty" as I left it. But it does'nt. It will restore to the last project I had opened instead of the bare, empty state. 
I do want to it to reopen to a project if I left it open. But I also want it to restore to empty if I consciously do so before closing.

**Comment by noinfo** [comment]
> Are you looking for something different than setting ` "restore_on_startup": "none"` in your settings.json?

I tried using this as a work around - even though it's not what I want and noticed something interesting (probably another issue). The setting works when launching from the macOS GUI, Spotlight or Alfred.app. The setting does not do anything at all when launching from the CLI. It launches a second window that will also reopen the last opened project. Opening the settings there will show that   ` "restore_on_startup": "none"` is correctly set however.

**Comment by williamviktorsson** [comment]
> > Are you looking for something different than setting ` "restore_on_startup": "none"` in your settings.json?
> 
> I tried using this as a work around - even though it's not what I want - and noticed something interesting (probably another issue). The setting works when launching from the macOS GUI, Spotlight or Alfred.app. The setting does not do anything at all when launching from the CLI. It launches a second window that will also reopen the last opened project. Opening the settings there will show that ` "restore_on_startup": "none"` is correctly set however.

This is actually a different issue: 
- #9592

I take it you want Zed to open with an empty project, should you have exited the application in the below displayed state?

<img width="667" alt="Screenshot 2024-04-25 at 09 49 55" src="https://github.com/zed-industries/zed/assets/22855292/e6243980-7c20-40ed-8f02-c6361dabf800">

**Comment by notpeter** [comment]
I can duplicate this with current releases. If you close the workspace (cmd+w) before quitting (cmd+q) when you come back, Zed reopens that workspace instead of giving you a blank window.

**Comment by lufog** [comment]
~On Linux,  ` "restore_on_startup": "none"`  setting does not work. The editor still opens the folder from the last session.~

In version 0.146.3 the setting works properly.

**Comment by Kan5661** [comment]
Was having the same issue here. Tried `"restore_on_startup": "none"` in settings.json and it did prevent previously worked on projects from opening. But the issue isn't entirely fixed because when opening up a project using the cli you should expect the project to be the only one that opens, however now I am getting the project I wanted opened as well as a blank project.

**Comment by perigozin** [comment]
I'm using Zed Dev 0.160.0 and I'm trying to understand this behavior.
I remove all folders from the project until the sidebar shows "Open a project Ctrl + O".
When I reopen Zed it shows the last folder I removed from the project no matter what I try.
`"restore_on_startup": "none"` shouldn't be needed because sometimes I specifically WANT for the folders to be restored but if I purposefully remove them, it should restore the same state as when I closed it.

In short: I expect Zed to have exactly the same state when reopened no matter if I have any folders added to a project. If none are opened upon closing the application it should reopen exactly as it was left.

**Comment by notpeter** [comment]
I cannot reproduce with these steps:
```
# zed is not open
mkdir -p /tmp/a /tmp/b
zed /tmp/a/
# cmd-q to quit zed
zed /tmp/b/
```
But others definitely can. Not sure what triggers this.

Here's a video of it happening to @stravo1 from https://github.com/zed-industries/zed/issues/9592#issuecomment-2993676208 on macOS 14.7.6, M1 Air, Zed 0.191.7 

<details><summary>video</summary>

https://github.com/user-attachments/assets/f635e569-6fc5-4610-9f8e-06be25fb4521

</details>

*shrug*

**Comment by ssergiienko** [comment]
@notpeter I have this/similar bug with steps to reproduce. For me its when I switch between local and remote.
STR:
1. Open local project
2. In the **same** window open remote project. You will end-up with only 1 window
4. Close the window (which will close Zed entirely)
5. Now I expect that when I open Zed again via desktop icon only last opened window(remote project) will be opened, but what happens instead is Zed opens 2 windows with both Local and Remote projects

The key is start from Local project. Next open/close does what is expected. Reproduces 100% times on Zed v0.195.3

My speculation is that in step 2 state is not updated to reflect that Local project window was closed(because it was implicit by switch to Remote) and it no longer needs to be reopened on start

**Comment by notpeter** [comment]
Thanks @ssergiienko. I am able to reproduce with your steps on macOS.

If I quit with the remote workspace open (cmd-q) it will restore the remote workspace when I reopen Zed (correct)

If I close the remote workspace (cmd-w), then quit Zed (cmd-q) it will restore the last local workspace, not the last closed workspace.

**Comment by smitbarmase** [comment]
I currently see two issues that should be fixed:

1. Closing the remote workspace (cmd-w), then quiting Zed (cmd-q) it will restore the last local workspace. This should not happen.
2. In local project, removing all folders from the project manually restores the last local folder on re-open. This should not happen.

I no longer can reproduce:
1. @Kan5661 mentioned, "Opening up a project using the cli you should expect the project to be the only one that opens, however now I am getting the project I wanted opened as well as a blank project."
2. @noinfo mentioned (original issue): "If you close the workspace (cmd+w) before quitting (cmd+q) when you come back, Zed reopens that workspace instead of giving you a blank window."

I'm on macOS, so it might be different on Linux/Windows will have to check.

**Comment by stravo1** [comment]
this issue is still there:

https://github.com/zed-industries/zed/issues/9592#issuecomment-2993676208
